hail/src/main/scala/is/hail/expr/ir/lowering/LowerBlockMatrixIR.scala

```diff
@@ -223,10 +223,10 @@ object LowerBlockMatrixIR {
         new BlockMatrixStage(Array(elt.name -> eltValue), TTuple(TInt64, TInt64)) {
           def blockContext(idx: (Int, Int)): IR = {
             val (i, j) = x.typ.blockShape(idx._1, idx._2)
-            MakeTuple.ordered(FastSeq(I64(i), I64(j)))
+            MakeTuple.ordered(FastSeq(I64(i.toInt), I64(j.toInt)))
           }
           def blockBody(ctxRef: Ref): IR =
-            MakeNDArray(ToArray(mapIR(rangeIR(GetTupleElement(ctxRef, 0) * GetTupleElement(ctxRef, 1)))(_ => elt)),
+            MakeNDArray(ToArray(mapIR(rangeIR( invoke("toInt32", TInt32, GetTupleElement(ctxRef, 0) * GetTupleElement(ctxRef, 1))) )(_ => elt)),
```

Comment:
is this doing a int64=>int32 cast? this should just be a `Cast` node then. You can also use the alias `(ir).toI`

Reply:
Fixed, didn't know about `Cast` node, much easier.